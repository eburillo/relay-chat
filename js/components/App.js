import React from 'react';
import Relay from 'react-relay';
import ItemCounter from './ItemCounter';
import Entry from './Entry/Entry';
import InputText from './InputText';
import AddEntryMutation from '../mutations/AddEntryMutation';

class App extends React.Component {
  render() {
    let entries = this.props.viewer.entryConnection.edges;
    return (
      <div className="app-container">
          <ItemCounter viewer={this.props.viewer}></ItemCounter>
          <ul className="entry-list">
            {entries.map(entry =>
              <Entry viewer={this.props.viewer} key={entry.node.id} entry={entry.node}></Entry>
            )}
          </ul>
          <InputText onSave={this._handleTextInputSave}></InputText>
      </div>
    );
  }

  _handleTextInputSave = (text) => {
    this.props.relay.commitUpdate(
      new AddEntryMutation({
        text,
        viewer: this.props.viewer
      }), {
        onSuccess: () => this.scrollDownList(),
        onFailure: () => { console.log('Add Entry Mutation failed!'); }
      }
    );
  };

  scrollDownList = function() {
    const elem = document.querySelector(".entry-list");
    var timeout = setTimeout(function() {
      elem.scrollTop += elem.scrollHeight;
    }, 200);
  }
}

export default Relay.createContainer(App, {
  initialVariables: {
    maxEntries: 50
  },
  fragments: {
    viewer: () => Relay.QL`
      fragment on User {
        id,
        entryConnection(last: $maxEntries) {
          edges {
            node {
              id,
              ${Entry.getFragment('entry')}
            }
          },
        },
        ${ItemCounter.getFragment('viewer')},
        ${AddEntryMutation.getFragment('viewer')},
      }
    `,
  },
});

// ${AddEntryMutation.getFragment('viewer')}, TODO: add to fragment
