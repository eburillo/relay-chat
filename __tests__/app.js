import React from 'react';
import Relay from 'react-relay';
import renderer from 'react-test-renderer';
import App from '../js/components/App';
import AppHomeRoute from '../js/routes/AppHomeRoute';

describe('Render App', function() {
  it('renders without crashing', () => {
    const tree = renderer.create(
      <Relay.Renderer
        environment={Relay.Store}
        Container={App}
        queryConfig={new AppHomeRoute()}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
