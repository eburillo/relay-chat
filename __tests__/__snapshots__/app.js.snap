// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App renders without crashing 1`] = `null`;
